libraries/pressureTransducer/ducer.h

```diff
@@ -114,18 +108,42 @@ namespace Ducers {
   }
 
   void readAllLowPressures(float *data) {
-    data[0] = ads1.readData(1) - calibration1;
-    data[1] = ads1.readData(0) - calibration1;
-    data[2] = ads1.readData(3) - calibration1;
-    data[3] = ads1.readData(2) - calibration1;
+    ads2.readData(0);
+    data[0] = ads1.readData(0);// - calibration1;
+    data[1] = ads1.readData(1);// - calibration1;
+    data[2] = ads1.readData(2);// - calibration1;
+    data[3] = ads1.readData(3);// - calibration1;
     data[4] = -1;
   }
 
   void readAllPressures(float *data) {
     readAllLowPressures(data);
-    data[4] = ads2.readData(0) - calibration2;
+    long temp = ads2.readData(0);
+    //Serial.println(temp);
+    data[4] = temp; //- calibration2;
     data[5] = -1;
   }
+
+
+      float tempOverVoltageScale = 1 / 0.01;
+      float voltageOffset = 0.75, tempOffset = 25; // 25 C = 0.450 V
+
+      long rawRead;
+      float voltageRead;
+      float tempRead;
+
+  void readTemperatureData(float *data) {
+    ads2.readData(1);
+    rawRead = ads2.readData(1); // thermocouple on AIN1 of ADC2
+    // Serial.println(rawRead);
+    // Serial.println(ads.readData(2));
+    // Serial.println(ads.readData(3));
+    voltageRead = (float) rawRead * (5.0 / pow(2,23));
+    // Serial.println(voltageRead);
+    tempRead = ((voltageRead - voltageOffset) * tempOverVoltageScale) + tempOffset;
+    data[0] = tempRead;
+    data[1] = -1;
+  }
```

Comment:
Putting this in this file is kind of sus. Had issues with two instances of the ADS object. Will change to create those objects in brain and pass to ducers and therm libs so that there is shared access to the same object. But for now, this is what we have.

Reply:
Sounds good, I just looked at the brain looks like this change has already been done?

Reply:
nope. definitely has not been done.